docs/ZipperTutorial.ml

```diff
@@ -0,0 +1,78 @@
+(* An introduction to top-down zippers for structured editing *)
+
+(* Given an algebraic data type, we can mechanically generate another type,
+   typically called the zipper or the type of 'one-holed contexts' relative
+   to the original type. Zippers come in top-down and bottom-up varieties.
+   Here we focus on top-down zippers, the variant used in Hazel.
+
+   Conceptually a value of this new type carries the same information
+   as a corresponding value of the original type, except that it also
+   distinguishes a particular subterm, which we call the 'cursor term'.
+
+   Given a zippered type, we can also mechanically define a corresponding
+   type of movement actions, each of which coresponds to a way of locally
+   changing which subterm is being indicated. *)
+
+(* A simplified AST type, representing unannotated binary trees *)
+type ast =
+  | Atom
+  | Bin of ast * ast
+
+(* A zippered representation of the above AST. In general, the zippered
+   version of a recursive ADT will have n variants for every n-ary
+   constructor (n-ary in the sense of containing n references to the
+   recursive type), plus an aditional variant representing the cursor
+   itself. So in our example, we have 0 variants for the 0-ary Atom,
+   2 variants for the binary Bin (representing the cursor being within
+   the left or right subtrees respectively), and the cursor case,
+   giving 3 variants in total *)
+type zast =
+ | Cursor of ast
+ | BinL of zast * ast
+ | BinR of ast * zast
+
```

Comment:
Maybe include an example or two here showing the cursor on a leaf but also on an interior node?